src/create-reminder-signup/lambda/lambda.ts

```diff
@@ -39,32 +32,46 @@ const identityAccessTokenPromise: Promise<string> = getParamFromSSM(
 	`/support-reminders/idapi/${ssmStage}/accessToken`,
 );
 
-export const run = async (event: SQSEvent): Promise<void> => {
+export const run = async (event: SQSEvent): Promise<SQSBatchResponse> => {
 	console.log('received event: ', event);
 
+	// SQS event source is configured with batchSize = 1
+	const record = event.Records[0];
+
+	// If there is an error, return the messageId to SQS
+	return processRecord(record)
+		.then(() => ({
+			batchItemFailures: [],
+		}))
+		.catch((error) => {
+			console.log(error);
+			return {
+				batchItemFailures: [{ itemIdentifier: record.messageId }],
+			};
+		});
+};
+
+const processRecord = (record: SQSRecord): Promise<void> => {
```

Comment:
functions in this file now use `Promise<void>` as the result, with any errors leading to a failed promise. The `run` function handles failure at the end